src/main/java/uk/gov/companieshouse/accounts/association/controller/UserCompanyAssociations.java

```diff
@@ -343,7 +348,16 @@ private void sendEmailNotificationForStatusUpdate(String xRequestId, String requ
         }
 
         if (authorisedUserRemoved) {
-            emailService.sendAuthorisationRemovedEmailToAssociatedUsers(xRequestId, companyDetails, requestingUserDisplayValue, targetUserDisplayValue, requestsToFetchAssociatedUsers);
+            final List<Supplier<User>> requestsToGetRecipientsOfAuthorisedUserRemovalEmail = new ArrayList<>(requestsToFetchAssociatedUsers);
```

Comment:
do we need this new arraylist supplier?
you could just add
requestsToFetchAssociatedUsers.add(requestToGetAssociationOwner);

Reply:
Makes sense, I'll refactor this as part of the work I'm doing just now.  So initially I'd created a new list to not affect the items in the existing one, as there are additional usages of it inside the function, so I didn't want to affect them - after reading more of the code, it seems the following statements:

https://github.com/companieshouse/accounts-association-api/blob/9b444696c5b9f6edb526ce0a035c57aedc21c14c/src/main/java/uk/gov/companieshouse/accounts/association/controller/UserCompanyAssociations.java#L334-L337

Are all mutually exclusive, so 4 isolates if statements could be if, if else etc.  I'd assumed they were split out for a reason but I guess not.  I'll refactor this so it's more clear anyways!

Reply:
Actually, we can't just call `.add` as `requestsToFetchAssociatedUsers` is immutable